src/wh_utils.c

```diff
@@ -84,3 +86,71 @@ int wh_Utils_memeqzero(uint8_t* buffer, uint32_t size)
     }
     return 1;
 }
+
+/** Cache helper functions */
+const void* wh_Utils_CacheInvalidate(const void* p, size_t n)
+{
+    int len = (int)n;
+    const uint8_t* ptr = (const uint8_t*)p;
+    do {
+        XCACHEINVLD(ptr);
+        ptr += XCACHELINE;
+        len -= XCACHELINE;
+    } while (len > 0);
+    return p;
+}
+
+void* wh_Utils_CacheFlush(void* p, size_t n)
+{
+    int len = (int)n;
+    uint8_t* ptr = (uint8_t*)p;
+    do {
+        XCACHEFLUSH(ptr);
+        ptr += XCACHELINE;
+        len -= XCACHELINE;
+    } while (len > 0);
+    return p;
+}
```

Comment:
I wonder if this should be a callback function left up to the port.... For example TC3xx lets you flush/invalidate the entire cache, a wordline, or a block, but the HSM cache is totally different.

Reply:
good call.  updated.